lib/classes/Swift/CharacterStream/NgCharacterStream.php

```diff
@@ -172,6 +172,10 @@ public function read($length)
                 $ret = substr($this->_datas,
                         $this->_currentPos * $this->_map,
                         $len);
+                // This check is necessary as of PHP 7.
+                if ($ret === '' && $this->_currentPos * $this->_map == strlen($this->_datas)) {
```

Comment:
We should probably also add a comment to explain why we need this piece of code.